### PR TITLE
added DOSCAR.LSO.lobster file attribute to Lobsterout, updated tests

### PR DESCRIPTION
**Pull request by naik-aakash**
## Lobsterout.get_doc() method updated

- Added has_doscar_lso: bool attribute to Lobsterout (checks if DOSCAR.LSO.lobster file is written during lobster run) 
- Updated corresponding tests

### CODE REVIEW

**Comment by naik-aakash** [comment]
Hi @janosh ,This PR is ready to be merged. Thanks !!

**Review by janosh — CHANGES_REQUESTED** [review]
Looks good! Just one comment about the test

File: pymatgen/io/lobster/tests/test_lobster.py [hunk 1]
```diff
@@ -1403,6 +1406,8 @@ def testattributes(self):
         self.assertEqual(self.lobsterout_cobi_madelung.has_cohpcar, True)
         self.assertEqual(self.lobsterout_cobi_madelung.has_madelung, True)
```
[the remark this answers is not in the document]
Comment (janosh) [reply pymatgen/io/lobster/tests/test_lobster.py:1408]: Please always use existing test files if possible.

[the remark this answers is not in the document]
Comment (naik-aakash) [reply pymatgen/io/lobster/tests/test_lobster.py:1409]: Thanks. I updated the test to test False on another file, as per your suggestion

**Comment by janosh** [comment]
Thanks!

**Pull request merged**